File: src/main/java/github/macro/build_info/gems/Vendor.java
package github.macro.build_info.gems;

import github.macro.build_info.ClassTag;

import java.util.SortedSet;

/**
 * Created by Macro303 on 2020-Jan-09.
 */
public class Vendor {
	private final String vendor;
	private final int act;
	private final String quest;
	private final SortedSet<ClassTag> classes;

	public Vendor(String vendor, int act, String quest, SortedSet<ClassTag> classes) {
		this.vendor = vendor;
		this.act = act;
		this.quest = quest;
		this.classes = classes;
	}

	public String getVendor() {
		return vendor;
	}

	public int getAct() {
		return act;
	}

	public String getQuest() {
		return quest;
	}

	public SortedSet<ClassTag> getClasses() {
		return classes;
	}

	@Override
	public String toString() {
		return "Vendor{" +
				"vendor='" + vendor + '\'' +
				", act=" + act +
				", quest='" + quest + '\'' +
				", classes=" + classes +
				'}';
	}
}

File: src/main/kotlin/github/macro/build_info/gems/Vendor.kt
package github.macro.build_info.gems

import github.macro.build_info.ClassTag
import javafx.beans.property.SimpleIntegerProperty
import javafx.beans.property.SimpleListProperty
import javafx.beans.property.SimpleStringProperty
import javafx.collections.FXCollections
import tornadofx.getValue
import tornadofx.setValue

/**
 * Created by Macro303 on 2020-Jan-13.
 */
class Vendor(
    vendor: String,
    act: Int,
    quest: String,
    classes: List<ClassTag>
) {
    val vendorProperty = SimpleStringProperty()
    var vendor by vendorProperty

    val actProperty = SimpleIntegerProperty()
    var act by actProperty

    val questProperty = SimpleStringProperty()
    var quest by questProperty

    val classesProperty = SimpleListProperty<ClassTag>()
    var classes by classesProperty

    init {
        this.vendor = vendor
        this.act = act
        this.quest = quest
        this.classes = FXCollections.observableList(classes)
    }

    override fun toString(): String {
        return "Vendor(vendor=$vendor, act=$act, quest=$quest, classes=$classes)"
    }
}